### PR TITLE
DBZ-3082 Provide Flight Recorder support

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-3082

### CODE REVIEW

File: connect-base/1.6/docker-entrypoint.sh [hunk 1]
```diff
@@ -125,6 +125,38 @@ if [[ -n "$JMXPORT" && -n "$JMXHOST" ]]; then
     export KAFKA_JMX_OPTS="-Djava.rmi.server.hostname=${JMXHOST} -Dcom.sun.management.jmxremote.rmi.port=${JMXPORT} -Dcom.sun.management.jmxremote.port=${JMXPORT} -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.authenticate=${JMXAUTH} -Dcom.sun.management.jmxremote.ssl=${JMXSSL} "
 fi
 
+#
+# Setup Flight Recorder
+#
+if [[ -n "$JFR_RUN_FILENAME" ]]; then
```
Comment (gunnarmorling) [inline connect-base/1.6/docker-entrypoint.sh:131]: @jpechane, filename isn't a mandatory option. You can start a continuous recording (which you then can dump using JMC for instance) solely via -XX:StartFlightRecording. I.e. we cannot rely on presence of JFR_RUN_FILENAME. How about this:

* Add an env var `ENABLE_JFR`, akin to `ENABLE_DEBEZIUM_SCRIPTING` and `ENABLE_APICURIO_CONVERTERS`
* If true (the default), the `-XX:StartFlightRecording` will be passed
* any variables prefixed with `JFR_RECORDING` (I think that's better than "RUN") or `JFR_OPT` will be passed through to `-XX:StartFlightRecording` and `-XX:FlightRecorderOptions`, respectively
* If there's at least one `JFR_OPT` variable, `-XX:FlightRecorderOptions` will be set (i.e. that's independent of `-XX:StartFlightRecording`)

That way, there'll be a continuous recording enabled by default, using default settings (e.g. max buffer size of 250 MB). WDYT?

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
@jpechane, so after some more thinking I believe we should add this. One specific suggestion on how to do it inline. And we'd want the same change for the Kafka image, too.

File: connect-base/1.6/docker-entrypoint.sh (hunk 1 above)
Comment (jpechane) [reply connect-base/1.6/docker-entrypoint.sh]: @gunnarmorling Done

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Applied, thx, @jpechane!

**Pull request merged**